mlflow/genai/judges/adapters/databricks_managed_judge_adapter.py

```diff
@@ -267,6 +276,86 @@ def _serialize_messages_to_databricks_prompts(
     return user_prompt, system_prompt
 
 
+def _run_databricks_agentic_loop(
+    messages: list["litellm.Message"],
+    trace: "Trace | None",
+    on_final_answer: Callable[[str | None], T],
+) -> T:
+    """
+    Run an agentic loop with Databricks chat completions.
+
+    This is the shared implementation for all Databricks-based agentic workflows
+    (judges, structured output extraction for traces). It handles the iterative
+    tool-calling loop until the LLM produces a final answer.
+
+    Args:
+        messages: Initial litellm Message objects for the conversation.
+        trace: Optional trace for tool calling. If provided, enables tool use.
+        on_final_answer: Callback to process the final LLM response content.
+            Receives the content string (or None if empty) and should return
+            the appropriate result type or raise an exception.
+
+    Returns:
+        Result from on_final_answer callback.
+
+    Raises:
+        MlflowException: If max iterations exceeded or other errors occur.
+    """
+    # Enable tool calling if trace is provided
+    tools = None
+    if trace is not None:
+        tools = [tool.get_definition() for tool in list_judge_tools()]
+
+    # Agentic loop: iteratively call LLM and execute tools until final answer
+    max_iterations = MLFLOW_JUDGE_MAX_ITERATIONS.get()
+    iteration_count = 0
+
+    while True:
+        iteration_count += 1
+        if iteration_count > max_iterations:
+            raise MlflowException(
+                f"Completion iteration limit of {max_iterations} exceeded. "
+                f"This usually indicates the model is not powerful enough to effectively "
+                f"analyze the trace. Consider using a more intelligent/powerful model. "
+                f"In rare cases, for very complex traces where a large number of completion "
+                f"iterations might be required, you can increase the number of iterations by "
+                f"modifying the {MLFLOW_JUDGE_MAX_ITERATIONS.name} environment variable.",
+                error_code=REQUEST_LIMIT_EXCEEDED,
+            )
+
+        # Serialize messages to Databricks format
+        user_prompt, system_prompt = _serialize_messages_to_databricks_prompts(messages)
+
+        llm_result = call_chat_completions(
+            user_prompt, system_prompt or "", tools=tools, model=_DATABRICKS_AGENTIC_JUDGE_MODEL
+        )
+
+        # Parse response
+        output_json = llm_result.output_json
+        if not output_json:
+            raise MlflowException(
+                "Empty response from Databricks judge",
+                error_code=INVALID_PARAMETER_VALUE,
```

Comment:
idt this is an `INVALID_PARAMETER_VALUE` - maybe we can just remove the error code